File: tkpdinbox/src/main/java/com/tokopedia/inbox/attachproduct/view/viewholder/CheckableInteractionListenerWithPreCheckedAction.java
package com.tokopedia.inbox.attachproduct.view.viewholder;

import com.tokopedia.abstraction.base.view.adapter.holder.BaseCheckableViewHolder;

/**
 * Created by Hendri on 26/02/18.
 */

public interface CheckableInteractionListenerWithPreCheckedAction extends BaseCheckableViewHolder.CheckableInteractionListener {
    boolean shouldAllowCheckChange(int position, boolean checked);
}

File: features/communication/attachproduct/src/main/java/com/tokopedia/attachproduct/view/viewholder/CheckableInteractionListenerWithPreCheckedAction.kt
package com.tokopedia.attachproduct.view.viewholder

import com.tokopedia.abstraction.base.view.adapter.holder.BaseCheckableViewHolder.CheckableInteractionListener

interface CheckableInteractionListenerWithPreCheckedAction : CheckableInteractionListener {

    fun shouldAllowCheckChange(position: Int, checked: Boolean): Boolean

}
